src/POData/UriProcessor/QueryProcessor/QueryProcessor.php

```diff
@@ -510,15 +510,28 @@ private function readSkipOrTopOption($queryItem, &$value)
     private function checkForEmptyQueryArguments()
     {
         $serviceHost = $this->service->getHost();
-        if (!is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_FILTER))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_EXPAND))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_INLINECOUNT))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_ORDERBY))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_SELECT))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_SKIP))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_SKIPTOKEN))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_TOP))
-        ) {
+        $items = [
+            ODataConstants::HTTPQUERY_STRING_FILTER,
+            ODataConstants::HTTPQUERY_STRING_EXPAND,
+            ODataConstants::HTTPQUERY_STRING_INLINECOUNT,
+            ODataConstants::HTTPQUERY_STRING_ORDERBY,
+            ODataConstants::HTTPQUERY_STRING_SELECT,
+            ODataConstants::HTTPQUERY_STRING_SKIP,
+            ODataConstants::HTTPQUERY_STRING_SKIPTOKEN,
+            ODataConstants::HTTPQUERY_STRING_TOP
+        ];
+
+        $allNull = true;
+        foreach ($items as $queryItem) {
+            $item = $serviceHost->getQueryStringItem($queryItem);
+            $currentNull = null === $item;
+            $allNull = ($currentNull && $allNull);
+            if (false === $allNull) {
+                break;
+            }
+        }
+
+        if (false === $allNull) {
```

Comment:
Scrutinizer originally caught these - but it's a good idea to make it BLEEDING OBVIOUS you're intending to compare bools to other bools